### PR TITLE
Update makefile

### PR DESCRIPTION
**Pull request by andrewfrench**
Screenshot 2024-01-03 at 11 01 57 PM

- Add Makefile help text
- Requires `v` to be set when bumping release version

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
 	@git push --tags
 	@poetry build
 	@poetry publish
+
+.DEFAULT_GOAL := help
+.PHONY: help
+help: ## Print Makefile help text.
+	@grep -E '^[a-zA-Z_\/-]+:.*?## .*$$' $(MAKEFILE_LIST) \
+	| awk 'BEGIN {FS = ":.*?## "}; \
+		{printf "\033[36m%-20s\033[0m%s\n", $$1, $$2}'
```
Comment (collindutter) [inline Makefile:22]: Can you explain this wizardry a bit 😅

Comment (andrewfrench) [reply Makefile:22]: This uses grep to parse the Makefile for targets with comments in the format:

```
target: ## comment
```

These are passed to awk which then splits the incoming line into a target and comment variable, then prints them inside some terminal formatting directives. Any targets without comments are ignored and excluded from help text.

**Review by collindutter — APPROVED** [review]

**Review by vasinov — APPROVED** [review]
Are there other checks we should add? For example, I think it currently fails if the git tag already exists (in the scenario where a previous run failed at the poetry build/publish step).

**Comment by andrewfrench** [comment]
I think there are -- it'd be cool if we could unify our test and static check entrypoints through the Makefile, giving us consistent behavior for testing pre-merge and with or without pre-commit hooks. I know we do at least some of these via marketplace actions so I can take a look later on to see how feasible that is.

**Pull request merged**